### DISCUSSION TITLE
How to trigger ci runs automatically ?

### DISCUSSION THREAD

**Discussion by rthangam** [GoogleChrome/lighthouse-ci#573] [General]
I have done the following:

- I installed npm modules for **lhci server** and **client.** 
- I have setup a configuration for server to run and upload to url in local host, it is working fine (I was able to setup version control tool so that it takes out latest hashes ). 
- I used patrickhulce/lhci-server to deploy to docker.

When i run `lhci autorun` it collects reports and uploads to the server.  How can i trigger it automatically whenever there is checkin to repository ? Does lighthouse-ci provides any support or i need to trigger it ?

**Comment by rthangam** [comment]
Can anyone please respond to this query ?

**Reply by patrickhulce** [reply]
It sounds like you don't have CI actually set up if you don't have something running on each commit already. You might want to take a look at the [docs for introduction to CI](https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/introduction-to-ci.md#setting-up-ci).

**Reply by rthangam** [reply]
Thanks  Patrick I have checked the mentioned links, we don't have them. We do have jenkins support is it possible to use CI in Jenkins ?

**Reply by patrickhulce** [reply]
Yes you can run Lighthouse CI in Jenkins, (see [docs](https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/getting-started.md#configure-lighthouse-ci)). How to setup a Jenkins job is beyond the scope of Lighthouse CI docs.

https://www.vogella.com/tutorials/Jenkins/article.html#setting-up-a-jenkins-job might help you there?

**Reply by rthangam** [reply]
Thanks Patrick
